FirebaseStorage/Sources/Storage.swift

```diff
@@ -283,10 +283,10 @@ import FirebaseCore
 
   /// A map of active instances, grouped by app. Keys are FirebaseApp names and values are
   /// instances of Storage associated with the given app.
-  private static var instances: [String: Storage] = [:]
+  private nonisolated(unsafe) static var instances: [String: Storage] = [:]
```

Comment:
Fixes <img width="587" alt="Screenshot 2024-08-01 at 1 31 49 PM" src="https://github.com/user-attachments/assets/496e1a93-54df-4643-8a0d-5096782c0e78">

This should be safe since our implementation looks solid.

Reply:
Had to revert as it was not Swift 5 compatible (see PR description writeup)